crates/iceberg/src/arrow/caching_delete_file_loader.rs

```diff
@@ -901,4 +986,50 @@ mod tests {
             result.err()
         );
     }
+
+    #[tokio::test]
+    async fn test_equality_delete_with_binary_type() {
```

Comment:
It seems this test duplicate with existing test a lot, is it possible to extend existing test with binary type?